src/components/content/order/formElements/TextInput.tsx

```diff
@@ -3,28 +3,56 @@
  * SPDX-FileCopyrightText: Huawei Inc.
  */
 
-import { Form, Input, Tooltip } from 'antd';
+import { Form, Input, Select, Tooltip } from 'antd';
 import { EyeInvisibleOutlined, EyeTwoTone, InfoCircleOutlined } from '@ant-design/icons';
-import { DeployParam, TextInputEventHandler } from './CommonTypes';
+import { DeployJsonSchema, DeployParam, TextInputEventHandler } from './CommonTypes';
 import { DeployVariable } from '../../../../xpanse-api/generated';
-import React from 'react';
-
+import React, { useState } from 'react';
+import { Rule } from 'rc-field-form/lib/interface';
 export function TextInput({
     item,
     onChangeHandler,
 }: {
     item: DeployParam;
     onChangeHandler: TextInputEventHandler;
 }): React.JSX.Element {
+    const ruleItems: Rule[] = [{ required: item.mandatory }];
+    let regExp: RegExp;
+    const [valueList, setValueList] = useState<string[]>([]);
+    const [isEnum, setIsEnum] = useState<boolean>(false);
+    const setValidator = (_: unknown, value: string) => {
+        if (!regExp.test(value)) {
+            return Promise.reject(new Error(` the provided input does not match the pattern! ${regExp}`));
+        }
+        return Promise.resolve();
+    };
+
+    for (const key in item.valueSchema) {
+        if (key === DeployJsonSchema.ENUM.valueOf()) {
+            setIsEnum(true);
+            setValueList(item.valueSchema[key] as unknown as string[]);
+            ruleItems.push(
+                { type: 'enum' },
+                {
+                    required: true,

```

Comment:
whether it is mandatory or no.. should also come from the deploy variable. It is nto mandatory always.